packages/hydrogen/src/optimistic-ui/optimistic-ui.tsx

```diff
@@ -0,0 +1,49 @@
+import {useFetchers} from '@remix-run/react';
+
+export function useOptimisticDataFromActions<T>(identifier: string) {
```

Comment:
It's unfortunate to me that this data gets serialized to the server, when it shouldn't need to. It's used only on the client side for optimistic UI. Maybe we should bring this up again with the Remix team. I remember they were talking about having some client-only data that could be associated with an action.